src/features/instance/operations/queries/getComponentFile.ts

```diff
@@ -0,0 +1,38 @@
+import { instanceClient } from '@/config/instanceClient';
+
+import { queryOptions } from '@tanstack/react-query';
+
+type GetComponentFileRequest = {
+	file: string;
+	project: string;
+};
+
+type GetComponentFileResponse = {
+	file: string;
+	project: string;
+	birthtime: string;
+	message: string;
+	mtime: string;
+	size: number;
+};
+
+function getComponentFileQuery(getComponentFileRequest: GetComponentFileRequest) {
+	return queryOptions({
+		queryKey: ['get_component_file'] as const,
```

Comment:
I recommend making the queryKey more precise -- for anything that can impact the contents. Otherwise, the cache won't be accurate! And re-fetching won't be accurate either. So... probably file and project? Arguably the instanceId should be a part of it too 🥳